### DISCUSSION TITLE
How to disable PDB in 0.27.1 operator ver (helm)?

### DISCUSSION THREAD

**Discussion by EsDmitrii** [strimzi/strimzi-kafka-operator#9731] [Q&A] [answered]
Hello everyone!
We are currently using kafka v2.8.0.
Unfortunately due to the specifics of the product and the approval steps for upgrading the underlying infrastructure, we cannot promptly update the current kafka.
That said, Kubernetes upgrades are more painless and we are now on v1.23.7.
So we are facing issues related to deprecated api for PDB (The policy/v1beta1 API version of PodDisruptionBudget is no longer served as of v1.25.).
How can we disable PDB or change api version from policy/v1beta1 to policy/v1 in v0.27.1 using helm?
Appreciate any helm!

**Comment by scholzj** [comment]
Strimzi 0.27 does not support Kubernetes 1.25 or higher. You have to updrade (you should have really done that before upgrading your Kubernetes cluster, but I think you should be able to upgrade it now without any major issues as well). You need at least Strimzi 0.29, but ideally you should use something much newer anyway.

**Comment by EsDmitrii** [comment]
I know but our apps depends on this critical infra, hundreds apps use kafka and we cant afford downtime when update it

**Reply by scholzj** [reply]
I don't think a Strimzi / Kafka upgrade means downtime. It involves the rolling of the pods etc. of course. But the same applies to the Kubernetes upgrade. So I think you should do the upgrade. And especially if this is a critical infra, it would bring also bug fixes, CVE fixes etc.

That said, if you do not want to upgrade, the only option is to edit the sources of Strimzi 0.27.1, remove the PDB management (that alone is not a huge change, you just comment out few lines of code), build a custom version os Strimzi cluster operator with it and use it in your cluster.

**Reply by EsDmitrii** [reply]
Can I kindly ask you to share these line with me, please? I don't understand java..

**Reply by scholzj** [reply, marked as answer]
For example this line: https://github.com/strimzi/strimzi-kafka-operator/blob/0.27.1/cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java#L327 => you can just comment it out and rebuild everything. There will be also similar for ZooKeeper: `zkPodDisruptionBudget`. So that one as well.

**Reply by EsDmitrii** [reply]
Thank you!

**Reply by EsDmitrii** [reply]
@scholzj hi!
Sorry to bother you
Can you share knowledge with me how to rebuild the project with my changes? 
I tried to rebuild several times and got different errors :(
Thank you in advance!

**Reply by scholzj** [reply]
The build details should be in the development docs: https://github.com/strimzi/strimzi-kafka-operator/blob/main/development-docs/DEV_GUIDE.md. Normally, you should install the prerequisites and do something like `MVN_ARGS=-DskipTests DOCKER_REGISTRY=<your-docker-registry> DOCKER_ORG=<your-docker-registry-user-or-org> make all`.
